src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separateDate = date.split(fromFormat.pop());
+
+  const dateParts = {
+    year: (fromFormat.includes('YYYY'))
+      ? separateDate[fromFormat.indexOf('YYYY')]
+      : separateDate[fromFormat.indexOf('YY')],
+    MM: separateDate[fromFormat.indexOf('MM')],
+    DD: separateDate[fromFormat.indexOf('DD')],
+  };
+
+  const toYear = (toFormat.includes('YYYY'))
+    ? 'YYYY'
+    : 'YY';
+
+  switch (true) {
+    case (toYear.length < dateParts.year.length):
```

Comment:
now it is totally unclear to me what happens in the code, some magic 😄
Let's create some variables for the conditions to make it more clear